snuba/migrate.py

```diff
@@ -42,9 +42,10 @@ def get_schema():
 
 
 def run(conn, dataset):
-    schemas = [dataset.get_dataset_schemas().get_read_schema()]
+    schemas = []
     if dataset.get_table_writer():
         schemas.append(dataset.get_table_writer().get_schema())
+    schemas.append(dataset.get_dataset_schemas().get_read_schema())
```

Comment:
running the read schema before the write schema for a matview would never work since the matview IS the read schema and the write schema is the underlying table.